juju/juju
# When an offer is consumed a second time, delete the old proxy

The offering model creates a proxy for the consuming app. If the consuming side does a `remove-saas` and then consumes the same offer again, the offering model might retain stale info, especially if it is down then the consuming side does a force delete.

This PR uses an incrementing version on the consuming side. If the offering side gets a new connection and the version is newer than what it has, it force deletes the consuming proxy and starts again. This is analogous to what is done on the consuming side then the offer is force removed and then consumed again.

This is same to do without a facade bump because the default version will be 0.

## QA steps

I tested this on top of #13353 but rebased against 2.9 to make this PR.

Deploy a cmr scenario like in #13349 and on the consuming side, remove the saas app and consume again.
Ensure relation hooks are run on both sides of the relation.

## Bug reference

https://bugs.launchpad.net/charm-ceph-mon/+bug/1940983
